src/app/modules/form/form.utils.ts

```diff
@@ -105,6 +105,19 @@ export const getFormFieldById = (
   return formFields.find((f) => fieldId === String(f._id)) ?? null
 }
 
+export const getFormFieldIndexById = (
+  formFields: IFormSchema['form_fields'],
+  fieldId: FormFieldSchema['_id'],
+): number | null => {
+  if (!formFields) {
+    return null
+  }
+
+  const index = formFields.findIndex((f) => fieldId === String(f._id))
```

Comment:
Why is the `String()` wrapper necessary here when it wasn't used in [useDuplicateFormField.ts](https://github.com/opengovsg/FormSG/pull/6285/files#diff-067e20b910f5477ea8e285ad392275bcca359c3af3032d8269792d1b88c529af)?

Reply:
Thats because the `f._id` here is a Mongoose `ObjectId` whilst the one in `useDuplicateFormField` is a `string` hahaha

And the reason why I had to wrap it in string instead of using `equals` comparator is because for some reason typescript or our current mongoose type does not recognise `ObjectId.equals` which is very weird 😩